app/models/conference/importer.rb

```diff
@@ -0,0 +1,74 @@
+require 'csv'
+class Conference::Importer
+  
+  ## This importer is depending on agreed-upon input format.
+  ## Input file:
+  # - Dates should be formatted with dd mm yyyy
+  # - UID should be unique and formatted with season + id: 2017001
+  # - The headers should not be changed
+  # Output:
+  # - Conferences will be updated or created
+  # - Conferences deleted in the input file, are not removed from the table
+  # - Import errors are logged in a Rails Logger
+  # - UID is mapped to 'gid' ('google-id' LOL), and has the format: 2017001
+  
+  class << self
+    def call(file)
+      check_valid(file)
+      info_log(:started, file)
+      process_csv(file)
+      info_log(:finished, file)
+    end
+    
+    private
+    
+    def info_log(arg, file)
+      notices = {
+        started: "Started importing file #{file.original_filename}",
+        finished: "Finished updating/creating #{count_conferences_in(file)} conferences"
+      }
+      logger.tagged("Importer") { logger.info notices[arg] }
+    end
+
+    def logger
+      Rails.logger
+    end
+    
+    def check_valid(file)
+      raise ArgumentError, "Oops! I can upload .csv only :-(" unless file.content_type == "text/csv"
+    end
+    
+    def count_conferences_in(file)
+      CSV.foreach(file.path, headers: true).count
+    end
+
+    def fetch_season_id(uid)
+      # uid format : 2017001
+      year = uid.to_s[0,4]
+      Season.find_or_create_by(name: year).id #Q: can I do that here?
```

Comment:
Sure, that should work. Is there a `!` version of `find_or_create_by`? If so, maybe use that one instead in case the first four characters of `uid` don't return a proper year.